app/models/schedule.rb

```diff
@@ -137,26 +137,43 @@ def clean_up_task_jobs
     end
   end
 
-  def timeline
-    self.root_task.self_and_descendants.collect do |task|
-      
-      unless task.sprout.nil?
-        rhizome_type = task.sprout.type.titlecase
-        number = task.sprout.rhizome_eid
-      end  
-
-      # this does not have the ramp up time for the thermostat included
-      [
-        "#{task.id}", 
-        "#{task.type_name_display} #{rhizome_type} #{number}", 
-        "#{rhizome_type}",
-        task.trigger_start*1000,
-        nil,
-        task.duration*1000,
-        nil,
-        ""  
-      ]
-    end 
+  # Provides a datatable representing the Schedule for use with a Google Chart Gantt chart
+  # @return [GoogleVisualr::DataTable] The datatable
+  def to_gantt_data
+    timeline = []
+    unless self.root_task.nil?
+      timeline = self.root_task.self_and_descendants.collect do |task|
+        task.to_gantt_data
+      end
+      timeline.flatten!(1)
+    end
+
+    chart_data = GoogleVisualr::DataTable.new
+    chart_data.new_column('string', 'Target ID')
+    chart_data.new_column('string', 'Task Name')
+    chart_data.new_column('string', 'Resource')
+    chart_data.new_column('date', 'Start Date')
+    chart_data.new_column('date', 'End Date')
+    chart_data.new_column('number', 'Duration')
+    chart_data.new_column('number', 'Percent Complete')
+    chart_data.new_column('string', 'Dependencies')
+    chart_data.add_rows(timeline)
+    chart_data
+  end
```

Comment:
The only issue I see with this at the moment is there are no hours on the x axis, it's just in ticks of ten minutes. This isn't a huge issue and I doubt there is an easy way to change it with the current google api anyway, but it be worth it to look into it for a few minutes.